### PR TITLE
Retain cell methods during spot extraction

### PR DESCRIPTION
**Pull request by gavinevans**
Description
This PR ensures that cell methods from the input gridded cube are retained on the output spot cube in the `SpotExtraction` plugin.

The output from one CLI acceptance test has been updated, where the input gridded cube had a cell method.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: improver_tests/spotdata/spotdata/test_SpotExtraction.py [hunk 1]
```diff
@@ -74,19 +74,25 @@ def setUp(self):
         # can be tested
         attributes = {"mosg__grid_domain": "global", "mosg__grid_type": "standard"}
 
+        cell_methods = (
+            iris.coords.CellMethod("maximum", coords="time", intervals="1 hour"),
+        )
+
```
Comment (bayliffe) [inline improver_tests/spotdata/spotdata/test_SpotExtraction.py:80]: Make this self.cell_methods, thus removing the need to redefine it in the test below.

File: improver_tests/spotdata/spotdata/test_SpotExtraction.py [hunk 2]
```diff
@@ -314,6 +320,20 @@ def test_cube_with_leading_dimensions(self):
...
+        expected_cell_methods = (
+            iris.coords.CellMethod("maximum", coords="time", intervals="1 hour"),
+        )
+        plugin = SpotExtraction(neighbour_selection_method="nearest_land")
+        result = plugin.process(
+            self.neighbour_cube,
+            self.diagnostic_cube_xy,
+            new_title="IMPROVER Spot Forecast",
+        )
+        self.assertEqual(result.cell_methods, expected_cell_methods)
```
Comment (bayliffe) [inline improver_tests/spotdata/spotdata/test_SpotExtraction.py:335]: Make this test against self.cell_methods

**Review by bayliffe — APPROVED** [review]
One small suggestion to remove some duplication from the tests, but happy that this does what is required.

File: improver_tests/spotdata/spotdata/test_SpotExtraction.py (hunk 1 above)
Comment (gavinevans) [reply improver_tests/spotdata/spotdata/test_SpotExtraction.py:80]: I've updated this as suggested.

File: improver_tests/spotdata/spotdata/test_SpotExtraction.py (hunk 2 above)
Comment (gavinevans) [reply improver_tests/spotdata/spotdata/test_SpotExtraction.py]: Updated.

**Review by MoseleyS — APPROVED** [review]

**Pull request merged**